### ISSUE TITLE
On the order page, customer relationships seem wrong.

### ISSUE THREAD

**Issue by marcomessa** [OFFLINE-GmbH/oc-mall-plugin#500]
Hi everyone, on the order page I happened to have this strange behavior:

<img width="1512" alt="Schermata 2020-05-05 alle 09 53 52" src="https://user-images.githubusercontent.com/4236142/81045475-913b5c00-8eb6-11ea-9411-2d93a4506b5d.png">

As you can see, emails are at first wrongly associated and then not displayed at all. I played around and found this.

```php
use Offline\Mall\Models\Order;

$o = Order::find(1);
$o->customer->user; // this is correct
$o->user; // this is wrong, it retrieves a wrong user.
```

Does anyone have this issue or I'm the only? In case, can this be related to Order hasManyThrough relationship?

```php
public $hasManyThrough = [
        'user'=> [
            User::class,
            'through'    => Customer::class,
            'throughKey' => 'id',
            'otherKey'   => 'customer_id',
            'key'        => 'user_id'
        ],
    ];
```

For now I solved overriding the column from a my plugin:

```php
// correggo bug utenti nella pagina degli ordini
        Event::listen('backend.list.extendColumns', function($widget) {
            if (!$widget->getController() instanceof Orders) {
                return;
            }
            if (!$widget->model instanceof Order) {
                return;
            }
            $widget->getColumn('user')->type = 'partial';
            $widget->getColumn('user')->path = '$/up/stella/overrides/_customer-email.htm';
        });
```

**Comment by tobias-kuendig** [comment]
Thank you very much for this report. It seems like the relationship definition is the culprit here.

The issue was introduced here to fix the search:
https://github.com/OFFLINE-GmbH/oc-mall-plugin/commit/e20ae57b02a43e036f3d6c8583ffb16d6cdaebff

The `$order->user` relationship should not exist in this case. The only safe way to access the user data is via `$order->customer->user`. 

The search issue has to be solved differently.

**Comment by tobias-kuendig** [comment]
Fixed in 486f6da, thanks again!

**Comment by marcomessa** [comment]
Ehi thank you for your quick reponse! And thank you for you incredible plugin, it's really awesome!
